### ISSUE TITLE
Code Insights - Unify Insights Tracking issue

### ISSUE THREAD

**Issue by chwarwick** [sourcegraph/sourcegraph#35387]
### Plan

Search-based Code Insights run either over an explicit list of repos or all repos. If the former, the data is generated by just-in-time searches. If the latter, the data is generated on a delay in the backend with results persisted in the database. These methods have different costs and benefits. We need to unify them to unblock developer velocity so we don’t have to build two versions of new features, and also to get the benefits of each across all insights while removing some of these costs.

see additional details in [rfc 677](https://docs.google.com/document/d/1ZBwHw8HqKVcOt_IOsSesC7lTadb5iJ_IY_RMg7kYq7M/edit#)
#### Legend

- 👩 Customer issue
- 🐛 Bug
- 🧶 Technical debt
- 🎩 Quality of life
- 🛠️ [Roadmap](https://handbook.sourcegraph.com/departments/product-engineering/process/planning-process#roadmap)
- 🕵️ [Spike](https://en.wikipedia.org/wiki/Spike_(software_development))
- 🔒 Security issue
- 🙆 Stretch goal

**Comment by sourcegraph-bot-2** [comment]
Heads up @joelkw @felixfbecker @vovakulikov @unclejustin - the "team/code-insights" label was applied to this issue.

**Comment by chwarwick** [comment]
@Joelkw I updated this with sub groups for each of the issues identified in [rfc 677](https://docs.google.com/document/d/1ZBwHw8HqKVcOt_IOsSesC7lTadb5iJ_IY_RMg7kYq7M/edit#).  You will notice there  is a lot of overlap on the `All search insights load equally fast after initial creation/backfill` and `All search insights can get additional data points and snapshots` items. I'll keep this up to date as we find and/or create new issues.

**Comment by Joelkw** [comment]
Thank you @chwarwick ! I will look this over more closely Wednesday – this is very helpful.

**Comment by Joelkw** [comment]
@chwarwick added the frontend live preview and pings issues here as well

**Comment by vovakulikov** [comment]
@chwarwick do we have an issue for tracking lang stats insight migration from the FE just in time to the backend GQL resolver?  

I know that I've been talking about it for quite some time but I want to mention it again, as long as we have to handle at least one  FE (just in time) insight on the frontend we still have a big tech complexity around insight types and they execution types. This means we have to care about all these FE local insights resolvers, different UI components on the dashboard and standalone insight pages. 

@Joelkw I can't answer the question - how much time it would solve for us in the next 2 months exactly. But I would say this is a very important and vital thing since we're going to have another insight type on the dashboard page (compute-powered) this means we only increase complexity but never simplify things on the FE, which means it would be easier for us to make a mistake on FE about data fetching, about filters flow, ...etc.

It also seems not optimal approach that we made a lot of work around unifying but still need to handle lang stats insight on the frontend. This means we got no benefits on the frontend from this big story and great things from @chwarwick in terms of tech FE complexity. Also moving Lang stats to the backend seems like a relatively easy ticket (@chwarwick correct me if I'm wrong here). So I think we should re-prioritize it. FE complexity is becoming more and more important if we want to integrate code insights UI with other parts of the product. 

cc @Joelkw @felixfbecker @chwarwick
